cmake/crc32.cmake

```diff
@@ -27,10 +27,22 @@ IF(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
     SET(CRC32_LIBRARY crc32_armv8_neon)
     ADD_SUBDIRECTORY(extra/crc32_armv8_neon)
   ENDIF()
-ENDIF()
 
-IF(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64")
+ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64")
   SET(HAVE_CRC32_VPMSUM 1)
   SET(CRC32_LIBRARY crc32-vpmsum)
   ADD_SUBDIRECTORY(extra/crc32-vpmsum)
+
+ELSEIF(NOT MSVC AND CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|i[0-9]86")
+  #Check for PCLMUL instruction (x86)
+  CHECK_C_SOURCE_COMPILES("
+  int main()
+  {
+    asm volatile (\"pclmulqdq \\$0x00, %%xmm1, %%xmm0\":::\"cc\");
+    return 0;
+  }"  HAVE_CLMUL_INSTRUCTION)
+  IF(HAVE_CLMUL_INSTRUCTION)
+    SET(CRC32_LIBRARY crc32-x86-pclmul)
+    ADD_SUBDIRECTORY(extra/crc32-x86-pclmul)
+  ENDIF()
```

Comment:
Instead of having extra/crc32-this and extra/crc32-that and multiple libraries , just have one and only extra/crc32 , put all sources there, and decide which *sources* files go into the one and only crc32 library based on IF(CMAKE_SYSTEM_PROCESSOR MATCHES)